### PR TITLE
Modified ofThe function (addresses follow-up for issue #2360)

### PR DESCRIPTION
**Pull request by Ant13731**
Changed all occurences of `ofThe` to `the_ofThe` for easier readability. Added in new `ofThe` function to work as expected.
closes issue #2360

### CODE REVIEW

**Comment by Ant13731** [comment]
Is the underscore the reason for the error here? Should I just use pure camel case?

**Comment by smiths** [comment]
Yes, @Ant13731, HLint failed on the camelCase test.  It seems that we could change `the_ofThe` to `theOfThe`.  I'm sure it is also possible to override the camelCase warning, but you would want permission from @JacquesCarette before going down that road.

**Comment by JacquesCarette** [comment]
If there's a way to make `the_ofThe` an exception to the camel case warning, that would be the best route. I like the underscore in that name. Otherwise, I think we'll turn off that warning.

**Comment by smiths** [comment]
There might be a hint on how to turn off the warning here:

https://github.com/haskell/haskell-ide-engine/issues/955

File: code/drasil-data/Data/Drasil/Concepts/Physics.hs [hunk 1]
```diff
@@ -118,10 +118,10 @@ pendulum = dccWDS "pendulum" (cn "pendulum")
 position = dcc "position" (cn' "position")
   "an object's location relative to a reference point"
 positionVec = dccWDS " positionVec" (cn' "position vector")
-   (S "a vector from the origin" `ofThe` phrase cartesian +:+ S "defined"
+   (S "a vector from the origin" `the_ofThe` phrase cartesian +:+ S "defined"
```
Comment (JacquesCarette) [inline code/drasil-data/Data/Drasil/Concepts/Physics.hs:121]: This one seems odd, i.e. wouln't it generate "the a vector..." ?

File: code/drasil-example/Drasil/GamePhysics/GenDefs.hs [hunk 2]
```diff
@@ -107,7 +107,7 @@ accelGravityDerivSentence2 = [S "The above equation governs the gravitational at
 
 accelGravityDerivSentence3 :: [Sentence]
 accelGravityDerivSentence3 =  [S "Given the above assumptions" `sC` S "let", ch mLarger `sAnd` ch QPP.mass, 
-        S "be the", phrase QPP.mass `ofThe` S "massive and light body respectively.",
+        S "be the", phrase QPP.mass `the_ofThe` S "massive and light body respectively.",
```
Comment (JacquesCarette) [inline code/drasil-example/Drasil/GamePhysics/GenDefs.hs:110]: won't this generate "the the" ?

**Review by JacquesCarette — CHANGES_REQUESTED** [review]

**Review by JacquesCarette — APPROVED** [review]

**Comment by JacquesCarette** [comment]
Except that your change to the hlint yaml seems to have a syntax error, which cased the build to fail. Otherwise, looks good.

File: code/.hlint.yaml [hunk 3]
```diff
@@ -54,6 +54,7 @@
 # Ignore some builtin hints
 # - ignore: {name: Use let}
 # - ignore: {name: Use const, within: SpecialModule} # Only within certain modules
+  - ignore: {name: "Use camelCase"}
```
Comment (balacij) [inline code/.hlint.yaml:57]: I believe you just need to remove the leading whitespace in front of the `-` so that it can be parsed for hlint.

Comment (Ant13731) [reply code/.hlint.yaml:57]: Ah okay I'll try that. Thanks!

Comment (smiths) [reply code/.hlint.yaml]: @Ant13731, you might want to run HLint before doing a pull request.  I think there is a make rule for hlint?

Comment (Ant13731) [reply code/.hlint.yaml]: make hlint didn't work but I think the program passed the build test for hlint at least.

**Pull request merged**